p256/point.ml

```diff
@@ -58,13 +60,18 @@ let to_affine p =
     Fe.to_bytes out_y y;
     Some (out_x, out_y)
 
-let pp fmt p =
+let to_cstruct p =
   match to_affine p with
-  | None ->
-      Format.fprintf fmt "00"
+  | None -> Cstruct.create 1
   | Some (x, y) ->
-      Format.fprintf fmt "04%a%a" Cstruct_util.pp_hex_le x
-        Cstruct_util.pp_hex_le y
+    let four = Cstruct.create 1 in
+    Cstruct.set_uint8 four 0 4 ;
+    let rev_x = Cstruct_util.rev x
+    and rev_y = Cstruct_util.rev y
+    in
+    Cstruct.concat [ four ; rev_x ; rev_y ]
+
+let pp fmt p = Cstruct.hexdump_pp fmt (to_cstruct p)
```

Comment:
I'd like to keep the format for now, so something like that should work (I'll upstream `Cstruct_util` and clean this mess later):
```suggestion
let pp fmt p = Cstruct_util.pp_hex_le fmt (Cstruct_util.rev p)
```